python/lsst/sims/catUtils/exampleCatalogDefinitions/phoSimCatalogExamples.py

```diff
@@ -9,14 +9,69 @@
 
 __all__ = ["write_phoSim_header", "PhosimInputBase",
            "PhoSimCatalogPoint", "PhoSimCatalogZPoint",
-           "PhoSimCatalogSersic2D", "PhoSimCatalogSSM", "PhoSimSpecMap"]
+           "PhoSimCatalogSersic2D", "PhoSimCatalogSSM", "PhoSimSpecMap",
+           "DefaultPhoSimHeaderMap"]
 
 
 PhoSimSpecMap = SpecMap(fileDict=defaultSpecMap.fileDict,
                         dirDict={'(^lte)': 'starSED/phoSimMLT'})
 
 
-def write_phoSim_header(obs, file_handle):
+# This is a dict of transformations mapping from data in OpSim to header
+# values expected by PhoSim.  The dict is keyed on the names of PhoSim
+# parameters.  The values of the dict are either straight values, in which
+# case those values are written to the PhoSim InstanceCatalog header, or tuples
+# containing the name of the OpSim column which should be use to calculate
+# the PhoSim parameter and any transformation needed to go from OpSim
+# units to PhoSim units.
+DefaultPhoSimHeaderMap = {'rottelpos': ('rotTelPos', np.degrees),
+                          'obshistid': ('obsHistID', None),
+                          'moonra': ('moonRA', np.degrees),
+                          'moondec': ('moonDec', np.degrees),
+                          'moonphase': ('moonPhase', None),
+                          'moonalt': ('moonAlt', np.degrees),
+                          'dist2moon': ('dist2Moon', np.degrees),
+                          'sunalt': ('sunAlt', np.degrees),
+                          'seeing': ('rawSeeing', None),
+                          'vistime': ('visitExpTime', lambda x: x+3.0),
+                          'nsnap': 2,
+                          'seed': ('obsHistID', None)}
+
+
+def evaluate_phosim_header(param, phosim_header_map, obs):
+    """
+    Find the value of a PhoSim header parameter given a PhoSimHeaderMap,
+    and an ObservationMetaData
+
+    @param [in] param is the name of the header parameter desired.
+
+    @param [in] phosim_header_map is a phoSimHeaderMap dict
+
+    @param [in] obs is an ObservationMetaData
+
+    @param [out] returns the value of the PhoSim header parameter
+    """
+
+    if param not in phosim_header_map:
+        return None
+
+    if isinstance(phosim_header_map[param], tuple):
+        transform = phosim_header_map[param]
+        if obs.OpsimMetaData is None:
+            return None
+
+        if transform[0] not in obs.OpsimMetaData:
+            return None
+
+        if transform[1] is None:
+            return obs.OpsimMetaData[transform[0]]
+
+        return transform[1](obs.OpsimMetaData[transform[0]])
```

Comment:
This is cool. I never thought about storing a numpy function like this to use later.